packages/flutter/lib/src/material/material_app.dart

```diff
@@ -68,30 +68,32 @@ class _MaterialAppState extends State<MaterialApp> {
   void initState() {
     super.initState();
     _navigator = new GlobalObjectKey(this);
-    WidgetFlutterBinding.instance.addEventListener(_backHandler);
     _size = ui.window.size;
-    FlutterBinding.instance.addMetricListener(_metricHandler);
+    FlutterBinding.instance.addObserver(this);
   }
 
   void dispose() {
-    WidgetFlutterBinding.instance.removeEventListener(_backHandler);
-    FlutterBinding.instance.removeMetricListener(_metricHandler);
+    FlutterBinding.instance.removeObserver(this);
     super.dispose();
   }
 
-  void _backHandler(InputEvent event) {
+  void onPushRoute(String route) {
```

Comment:
We've been using onFoo() for the fields you set, and handleFoo() for the handlers.